File: raspberry/SmartHome/app/src/main/java/ru/smarthome/server/http/handlers/BaseRequestHandler.java
package ru.smarthome.server.http.handlers;

import android.support.annotation.NonNull;

import java.util.Map;

import fi.iki.elonen.NanoHTTPD;
import ru.smarthome.library.BaseController;
import ru.smarthome.model.RaspberrySmartHome;

import static fi.iki.elonen.NanoHTTPD.MIME_PLAINTEXT;

public abstract class BaseRequestHandler implements RequestHandler{

    protected BaseController getController(Map<String, String> params) {
        // todo add checks so it won't crash
        long controllerGuid = Long.parseLong(params.get("controller_guid"));
        return RaspberrySmartHome.getInstance().getController(controllerGuid);
    }

    @NonNull
    protected NanoHTTPD.Response getArduinoHttpError() {
        return new NanoHTTPD.Response(NanoHTTPD.Response.Status.INTERNAL_ERROR, MIME_PLAINTEXT, "arduino web server not available");
    }

}

File: raspberry/SmartHome/arduinodevices/src/main/java/smarthome/raspberry/arduinodevices/server/httphandlers/BaseRequestHandler.kt
package smarthome.raspberry.arduinodevices.server.httphandlers

import fi.iki.elonen.NanoHTTPD
import fi.iki.elonen.NanoHTTPD.MIME_PLAINTEXT
import smarthome.library.common.BaseController
import smarthome.raspberry.arduinodevices.server.DeviceChannelInput
import smarthome.raspberry.arduinodevices.server.WebServerOutput

internal abstract class BaseRequestHandler : RequestHandler {

    protected val output: WebServerOutput = TODO()
    protected val input: DeviceChannelInput = TODO()

    protected val arduinoHttpError: NanoHTTPD.Response
        get() = NanoHTTPD.Response(NanoHTTPD.Response.Status.INTERNAL_ERROR, MIME_PLAINTEXT, "arduino web server not available")

    protected fun getInvalidRequestResponse(message: String): NanoHTTPD.Response {
        return NanoHTTPD.Response(NanoHTTPD.Response.Status.BAD_REQUEST, MIME_PLAINTEXT, "Invalid request: $message")
    }

    protected fun getController(params: Map<String, String>): BaseController {
        // todo add checks so it won't crash
        val controllerGuid = params.getValue("controller_guid").toLong()
        return input.findController(controllerGuid)
    }

}
